src/Symfony/Component/String/BinaryString.php

```diff
@@ -0,0 +1,398 @@
+
+        return $str;
+    }
+
+    public function prepend(string ...$prefix): parent
+    {
+        $str = clone $this;
+        $str->string = (1 >= \count($prefix) ? $prefix[0] ?? '' : implode($prefix)).$str->string;
+
+        return $str;
+    }
+
+    public function replace(string $from, string $to): parent
+    {
+        $str = clone $this;
+        $str->string = $this->ignoreCase ? str_ireplace($from, $to, $this->string) : str_replace($from, $to, $this->string);
+
+        return $str;
+    }
+
+    public function replaceMatches(string $from, $to): parent
+    {
+        if ('g' === $pattern[-1]) {
+            $pattern = substr($pattern, 0, -1);
+        }
+
+        if ($this->ignoreCase) {
+            $pattern .= 'i';
+        }
+
+        $replace = \is_array($to) || $to instanceof \Closure ? 'preg_replace_callback' : 'preg_replace';
+        set_error_handler(static function ($t, $m) { throw new InvalidArgumentException($m); });
+
+        try {
+            if (null === $string = $replace($from, $to, $this->string)) {
+                $lastError = preg_last_error();
+
+                foreach (get_defined_constants(true)['pcre'] as $k => $v) {
+                    if ($lastError === $v && '_ERROR' === substr($k, -6)) {
+                        throw new RuntimeException('Matching failed with '.$k.'.');
+                    }
+                }
+
+                throw new RuntimeException('Matching failed with unknown error code.');
+            }
+        } finally {
+            restore_error_handler();
+        }
+
+        $str = clone $this;
+        $str->string = $string;
+
+        return $str;
+    }
+
+    public function slice(int $start = 0, int $length = null): parent
+    {
+        $str = clone $this;
+        $str->string = (string) substr($this->string, $start, $length ?? PHP_INT_MAX);
+
+        return $str;
+    }
+
+    public function splice(string $replacement, int $start = 0, int $length = null): parent
+    {
+        $str = clone $this;
+        $str = substr_replace($this->string, $replacement, $start, $length ?? PHP_INT_MAX);
+
+        return $str;
+    }
+
+    public function split(string $delimiter, int $limit = null): array
+    {
+        if ('' === $delimiter) {
+            return $this->chunk();
+        }
+
+        $chunks = $this->ignoreCase
+            ? preg_split('{'.preg_quote($delimiter).'}i', $this->string, $limit ?? PHP_INT_MAX)
+            : explode($delimiter, $this->string, $limit ?? PHP_INT_MAX);
+
+        foreach ($chunks as $i => $chunk) {
+            $chunks[$i] = $str = clone $this;
+            $str->string = $chunk;
+        }
+
+        return $chunks;
+    }
+
+    public function startsWith(string $prefix): bool
+    {
+        if ('' === $prefix) {
+            return true;
+        }
+
+        return 0 === ($this->ignoreCase ? stripos($this->string, $prefix) : strpos($this->string, $prefix));
+    }
+
+    public function toCamelCase(bool $lcFirst): parent
```

Comment:
toCamelCase + toSnakeCase seems missing in the other implementations ...

Reply:
On purpose: you don't use snake/camel on Unicode strings. We could migrate to AbstractString, but I would wait for someone with an actual real-world use case before doing so.